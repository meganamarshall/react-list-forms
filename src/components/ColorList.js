import React, { PureComponent } from 'react';
import Colors from './Colors';
import CreateColor from './CreateColor';

export default class ColorList extends PureComponent {
  state = {
    colors: []
  }

  addColor = color => {
    this.setState(state => {
      return {
        colors: [...state.colors, color]
      };
    });
  }

  render() {
    const colors = this.state;
    return (
      <>
        <CreateColor />
        <Colors colors={colors} />
      </>
    );
  }
}
