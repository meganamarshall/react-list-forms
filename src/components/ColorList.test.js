import React from 'react';
import { shallow } from 'enzyme';
import ColorList from './ColorList';

describe('colorlist component', () => {
  it('renders the colorlist', () => {
    const wrapper = shallow(<ColorList />);
    expect(wrapper).toMatchSnapshot();
  });
});
