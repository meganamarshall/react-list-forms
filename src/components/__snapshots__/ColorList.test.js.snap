// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`colorlist component renders the colorlist 1`] = `
<Fragment>
  <CreateColor
    addColor={[Function]}
  />
  <Colors
    colors={Array []}
  />
</Fragment>
`;
